### PR TITLE
Upgrade to CI-based @folio/eslint-config-stripes

### PR DESCRIPTION
**Pull request by doytch**

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 # Ignore app templates from CI linting of stripes-cli due to template syntax
 # Run lint-templates to lint the app templates separately
 resources/**/*
+test
```
Comment (ttamj) [inline .eslintignore]: We may want to think about a separate configuration for tests.  See what started here: I https://github.com/folio-org/stripes-cli/blob/master/test/.eslintrc

**Comment by doytch** [comment]
@ttamj this is ready on my end if you're fine with merging with these current linter errors.

**Comment by ttamj** [comment]
Give me a few minutes and I'll clean it up.

**Comment by ttamj** [comment]
I cleaned up the errors and left the warnings for later.  In the process I disabled `prefer-promise-reject-errors` for tests only.  I did this to minimize complexity in creating stubs.  Any objections?

**Comment by ttamj** [comment]
Oh looks like I missed one... I'll clean it up next time I make a PR.

**Comment by doytch** [comment]
Yeah I think your thinking on Promises makes sense. Let's do it!

**Pull request merged**